### PR TITLE
Fully support server-generated resources.

### PR DESCRIPTION
**Pull request by kzemek**

### CODE REVIEW

File: src/escalus_utils.erl [hunk 1]
```diff
@@ -163,6 +164,9 @@ get_username(UserOrClient) ->
 get_server(UserOrClient) ->
```
Comment (gadgetci) [inline src/escalus_utils.erl]: According to **Compiler**:
> Warning: missing specification for function get_server/1

File: src/escalus_utils.erl [hunk 2]
```diff
@@ -163,6 +164,9 @@ get_username(UserOrClient) ->
...
     regexp_get(get_short_jid(UserOrClient), <<"^[^@]*[@]([^/]*)">>).
 
+get_resource(UserOrClient) ->
+    regexp_get(get_short_jid(UserOrClient), <<"^[^@]+@[^/]+/(.*)$">>).
+
 is_prefix(Prefix, Full) when is_binary(Prefix), is_binary(Full) ->
```
Comment (gadgetci) [inline src/escalus_utils.erl]: According to **Compiler**:
> Warning: missing specification for function is_prefix/2

File: src/escalus_utils.erl [hunk 3]
```diff
@@ -163,6 +164,9 @@ get_username(UserOrClient) ->
 get_server(UserOrClient) ->
     regexp_get(get_short_jid(UserOrClient), <<"^[^@]*[@]([^/]*)">>).
 
+get_resource(UserOrClient) ->
```
Comment (gadgetci) [inline src/escalus_utils.erl]: According to **Compiler**:
> Warning: missing specification for function get_resource/1

Comment (michalwski) [inline src/escalus_utils.erl]: Agree with @gadgetci here

File: src/escalus_utils.erl [hunk 4]
```diff
@@ -157,12 +158,18 @@ jid_to_lower(Jid) ->
...
 
+-spec get_server(UserOrClient :: jid_spec()) -> binary().
 get_server(UserOrClient) ->
     regexp_get(get_short_jid(UserOrClient), <<"^[^@]*[@]([^/]*)">>).
 
+-spec get_resource(JID :: binary()) -> binary().
+get_resource(JID) ->
+    regexp_get(JID, <<"^[^/]+/(.*)$">>).
+
 is_prefix(Prefix, Full) when is_binary(Prefix), is_binary(Full) ->
```
Comment (gadgetci) [inline src/escalus_utils.erl]: According to **Compiler**:
> Warning: missing specification for function is_prefix/2

File: src/escalus_client.erl [hunk 5]
```diff
@@ -141,8 +145,9 @@ username(Client) ->
 server(Client) ->
```
Comment (gadgetci) [inline src/escalus_client.erl]: According to **Compiler**:
> Warning: missing specification for function server/1

**Pull request merged**